### ISSUE TITLE
VideoCAD pretrained checkpoint — 80% CMD vs Table 3's 98%

### ISSUE THREAD

**Issue by HY666-UX** [ghadinehme/VideoCAD#11]
Hi Ghadi,

Thanks a lot for open-sourcing VideoCAD — the dataset and code are very well organized.

I evaluated the pretrained checkpoint (best_model.pt from Google Drive) on 1000 test samples using the default config (cad_past_10_actions_and_states_timestep_embedding, teacher forcing mode), and got:

  CMD accuracy:    80.34%   (Table 3: 98.08%)
  Params accuracy:  52.56%   (Table 3: 82.35%)

I noticed the Adam optimizer step in the checkpoint is ~385K, while a full 1000-epoch run on 41K samples with batch_size=2 would reach roughly 10–20 million steps. This suggests the uploaded checkpoint may be an intermediate snapshot rather than the fully converged model used for Table 3.

Could you confirm whether the Google Drive checkpoint is the same one that produced the Table 3 numbers? If not, would it be possible to share the final weights, or let me know when they become available?

Thanks again,
yhong

**Comment by ghadinehme** [comment]
Hi Yhong,

Thanks for reaching out and for testing VideoCAD. I double-checked the uploaded checkpoint, and it achieves the performance reported in the paper.

Could you please double-check your evaluation setup and configuration? There may be an issue on your end causing the discrepancy, since the reported accuracies are substantially lower than what I obtain with the same checkpoint.

Best,
Ghadi

**Comment by HY666-UX** [comment]
Dear Ghad,

I have resolved this reproduction issue on my end, thank you. Thank you for releasing VideoCAD. We are conducting research on world-model-guided precision action prediction for CAD GUI agents using VideoCAD.

We are now working to reproduce the results presented in Table 4. We would greatly appreciate it if you could provide the 200 test sample IDs used in this table, the action sequences predicted by VideoCADFormer, along with the corresponding Onshape executor and Chamfer distance evaluation code.

In addition, we wonder if you could share either:

The Selenium/PyAutoGUI/Xvfb data-generation or action-replay code used for VideoCAD, together with its environment configuration; or

A minimal runnable replay package for approximately 20–50 training episodes.

The most useful minimal package would include, where available:

Exact commit versions of the collector/replayer and their dependencies;

Chrome, ChromeDriver, Selenium, PyAutoGUI, Xvfb, viewport, DPI, and browser zoom settings;

The mapping between DeepCAD IDs, VideoCAD IDs, and Onshape document/workspace/element/version or microversion IDs;

A deterministic method for replaying a trajectory prefix from a blank document;

Restoration or replay of the active tool, constraint state, selection, focus, and camera;

The coordinate conversion between recorded pixels and the released VideoCAD action bins;

An observable result after executing a candidate action, such as a settled screenshot, hit/snap/entity information, feature or sketch diff, or error status.

Our first step would be a small train-only reproducibility test. We will not access the test set or launch large-scale automation. We are also happy to use private research access, sign an appropriate data-use agreement, and follow Onshape’s terms of service and rate limits.

If the original code cannot be shared, a short call, a minimal demonstration, an environment manifest, or confirmation of which state identifiers were not retained would still be extremely helpful.

Thank you very much for your time and for making VideoCAD available.
